src/jwe/index.ts

```diff
@@ -0,0 +1,76 @@
+import { JWEInvalid } from '../utils/errors';
+import type { JWEHeaderParameters, JWK } from '../types';
+import { IoReactNativeJwt } from '../utils/proxy';
+
+/**
+ * The EncryptJwe class is used to build and encrypt Flattened JWE objects.
+ *
+ * @example Usage
+ *
+ * ```js
+ * const jwe = await new EncryptJwe(
+ *   new TextEncoder().encode('It’s a dangerous business, Frodo, going out your door.'),
+ * )
+ *   .setProtectedHeader({ alg: 'RSA-OAEP-256', enc: 'A256GCM' })
+ *   .setAdditionalAuthenticatedData(encoder.encode('The Fellowship of the Ring'))
+ *   .encrypt(publicKey)
+ *
+ * console.log(jwe)
+ * ```
+ */
+export class EncryptJwe {
+  private _plaintext: String;
+
+  private _protectedHeader!: JWEHeaderParameters;
+
+  /** @param plaintext String representation of the plaintext to encrypt. */
+  constructor(plaintext: String) {
+    this._plaintext = plaintext;
+  }
+
+  /**
+   * Sets the JWE Protected Header on the EncryptJwe object.
+   *
+   * @param protectedHeader JWE Protected Header.
+   */
+  setProtectedHeader(protectedHeader: JWEHeaderParameters) {
+    if (this._protectedHeader) {
+      throw new TypeError('setProtectedHeader can only be called once');
+    }
+    this._protectedHeader = protectedHeader;
+    return this;
+  }
```

Comment:
*suggestion*: 
1. we must call it once before the signature
2. we must not call it more the once

Why don't just require this parameter in the constructor? That would also avoid all the null checks below.

Reply:
Great suggestion. Fixed!